{% extends "layout.html" %}
{% block body %}
    <script>
        /**
        * Function that tracks a click on an outbound link in Analytics.
        * This function takes a valid URL string as an argument, and uses that URL string
        * as the event label. Setting the transport method to 'beacon' lets the hit be sent
        * using 'navigator.sendBeacon' in browser that support it.
        */
        var trackOutboundLink = function(url) {

            ga('send', 'event', 'outbound', 'click', url, {
                'transport': 'beacon',
                'hitCallback': function(){document.location = url;}
            });
        }
    </script>
    <div class="container" style="font-size: 16px;">
        <h1 class="download-header">Downloads</h1>
        <!-- <p>
            Please subscribe to <a href="https://groups.google.com/forum/#!forum/exac_data_announcements" >our mailing list</a>
            to be notified when ExAC v2 and gnomAD VCFs are released.
        </p> -->
        <!-- <p>
            Data for release from the Genome Aggregation Database (gnomAD) consortium are available via FTP <a href="ftp://ftp.broadinstitute.org/pub/ExAC_release" >here</a> <em>(updated 02-15-2016)</em> or through the links provided below.
        </p> -->
        <p class="download-header">
            Genome Aggregation Database (gnomAD) data are available at the links below. Please see <a href="https://macarthurlab.org/2017/02/27/the-genome-aggregation-database-gnomad" >this blog post</a> for more information about this release.
        </p>
        <p class="download-header">
            Last updated <em>October 3, 2017 11:29 AM: release gnomAD version 2.0.2.</em>
        </p>
        <h3 class="download-header">Variant Dataset (VDS) format</h3>
        <p class="download-header">
            <a href="https://hail.is/" >Hail</a> users can download gnomAD data in <a href="https://hail.is/hail/overview.html#overview-vds" >Variant Dataset (VDS) format</a>.
        </p>
        <p class="download-header"> First, install <a href="https://cloud.google.com/sdk/docs/" >Google Cloud SDK</a>. Then use the <a href="https://cloud.google.com/storage/docs/gsutil" >Google Cloud Storage</a> <code>gsutil</code> tool to copy partitions in parallel to your own storage bucket or local hard drive. <a href="http://discuss.hail.is/t/using-hail-on-the-google-cloud-platform/80" >This blog post</a> can help you get started with Hail on the cloud and <a href="http://discuss.hail.is/t/using-filter-variants-intervals-to-query-the-16gb-gnomad-sites-dataset-in-300ms-on-your-laptop/148" >
            this post</a> shows you how to query variants by interval on your laptop.
        </p>
        <pre class="download-code"><code>mkdir gnomad_data</code></pre>
        <p class="download-header">
            To download VDS files <em>(note: autosomes are now combined with sex chromosomes)</em>:
        </p>
        <pre class="download-code"><code>gsutil -m cp -r gs://gnomad-public/release/2.0.2/vds/exomes/gnomad.exomes.r2.0.2.sites.vds gnomad_data # 16 GB
          gsutil -m cp -r gs://gnomad-public/release/2.0.2/vds/exomes/gnomad.exomes.r2.0.2.sites.split.vds gnomad_data
gsutil -m cp -r gs://gnomad-public/release/2.0.2/vds/genomes/gnomad.genomes.r2.0.2.sites.vds gnomad_data # 108 GB</code></pre>
        <p class="download-header">
        The original ExAC dataset is available here:
        </p>
        <pre class="download-code"><code>gsutil -m cp -r gs://gnomad-public/legacy/exac_browser/ExAC.r1.sites.vds gnomad_data # 6.7 GB</code></pre>

        <h3 class="download-header">VCF and coverage files</h3>
        <p class="download-header">
            Files are linked in the table below, or you can explore the data directory <a href="https://console.cloud.google.com/storage/browser/gnomad-public/release/">here</a> . To download multiple files in parallel on command line, follow these instructions: <p class="download-header"> First, install <a href="https://cloud.google.com/sdk/docs/" >Google Cloud SDK</a> (you will need a Google account). Then use the <a href="https://cloud.google.com/storage/docs/gsutil" >Google Cloud Storage</a> <code>gsutil</code> tool to copy files in parallel to your own storage bucket or local hard drive. Here are some example commands for interacting with the storage bucket:
        </p>
        <pre class="download-code"><code>gsutil ls gs://gnomad-public/release/2.0.2
gsutil ls gs://gnomad-public/release/2.0.2/vcf/genomes</code></pre>
        <p class="download-header">View file sizes:</p>
        <pre class="download-code"><code>gsutil du -h -s gs://gnomad-public/release/2.0.2/vcf/genomes # size of whole folder
gsutil du -h gs://gnomad-public/release/2.0.2/vcf/genomes # sizes of individual files</code></pre>
        <p class="download-header">To download, find somewhere to put the data.</p>
        <pre class="download-code"><code>mkdir gnomad_data</code></pre>
        <p class="download-header">
            <p class="download-header">
             Download all vcf files:
            </p>
            <pre class="download-code"><code>gsutil -m cp -r gs://gnomad-public/release/2.0.2/vcf gnomad_data # 102 GB</code></pre>
        </p>
        <p class="download-header">
         Or specific chromosomes:
        </p>
        <p class="download-header">
            <pre class="download-code"><code># e.g. chromosome 1, 7 GB
gsutil -m cp -r gs://gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr1.vcf.bgz gnomad_data</code></pre>
        </p>

        <!-- <p class="download-header">Explore the <a href="https://data.broadinstitute.org/gnomAD/release/2.0.2" >data directory</a> to understand the folder structure.</p> -->
         <ul class="media-list">
            <div class="media-body">
                <div class="download-tables">
                    <div class="download-table-group">
                        <h4>Exome Data</h4>
                        <table class="downloads-table" border="1" cellspacing="0" cellpadding="0">
                            <tbody align="center">
                                <tr>
                                    <td width="429 ">
                                        <p>
                                            <strong>Exome VCF Files</strong>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                     <td width="428">
                                         <p>
                                             <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/exomes/gnomad.exomes.r2.0.2.sites.vcf.bgz" >All chromosomes sites VCF (8.9 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/exomes/gnomad.exomes.r2.0.2.sites.vcf.bgz.tbi" >(.tbi)</a>
                                         </p>
                                     </td>

                            </tbody>
                        </table>
                        <table class="downloads-table" border="1" cellspacing="0" cellpadding="0">
                            <tbody align="center">
                                <tr>
                                    <td width="429 ">
                                        <p>
                                            <strong>Exome Coverage Files</strong>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr1.coverage.txt.gz" >chr 1 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr2.coverage.txt.gz" >chr 2 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr3.coverage.txt.gz" >chr 3 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr4.coverage.txt.gz" >chr 4 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr5.coverage.txt.gz" >chr 5 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr6.coverage.txt.gz" >chr 6 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr7.coverage.txt.gz" >chr 7 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr8.coverage.txt.gz" >chr 8 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr9.coverage.txt.gz" >chr 9 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr10.coverage.txt.gz" >chr 10 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr11.coverage.txt.gz" >chr 11 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr12.coverage.txt.gz" >chr 12 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr13.coverage.txt.gz" >chr 13 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr14.coverage.txt.gz" >chr 14 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr15.coverage.txt.gz" >chr 15 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr16.coverage.txt.gz" >chr 16 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr17.coverage.txt.gz" >chr 17 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr18.coverage.txt.gz" >chr 18 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr19.coverage.txt.gz" >chr 19 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr20.coverage.txt.gz" >chr 20 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>

                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr21.coverage.txt.gz" >chr 21 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chr22.coverage.txt.gz" >chr 22 coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chrX.coverage.txt.gz" >chr X coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/exomes/gnomad.exomes.r2.0.2.chrY.coverage.txt.gz" >chr Y coverage</a>
                                        </p>
                                    </td>
                                </tr>
                                 <tr>
                                    <td width="428">
                                        <p>
                                            <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/combined_tars/gnomad.exomes.r2.0.2.coverage.all.tar" >All exome coverage (1.1 GB)</a>
                                        </p>
                                    </td>
                                </tr>
                            </tbody>
                        </table>
                    </div>
                    <div class="download-table-group">
                        <h4>Genome Data</h4>
                        <table class="downloads-table" border="1" cellspacing="0" cellpadding="0">
                        <tbody align="center">
                            <tr>
                                <td width="429 ">
                                    <p>
                                        <strong>Genome VCF Files</strong>
                                    </p>
                                </td>
                                <td width="429">
                                    <p>
                                        <strong>Genome Coverage Files</strong>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr1.vcf.bgz" >chr 1 sites VCF (6.9 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr1.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr1.coverage.txt.gz" >chr 1 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr2.vcf.bgz" >chr 2 sites VCF (7.3 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr2.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr2.coverage.txt.gz" >chr 2 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr3.vcf.bgz" >chr 3 sites VCF (6.1 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr3.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr3.coverage.txt.gz" >chr 3 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr4.vcf.bgz" >chr 4 sites VCF (5.7 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr4.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr4.coverage.txt.gz" >chr 4 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr5.vcf.bgz" >chr 5 sites VCF (5.4 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr5.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr5.coverage.txt.gz" >chr 5 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr6.vcf.bgz" >chr 6 sites VCF (5.0 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr6.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr6.coverage.txt.gz" >chr 6 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr7.vcf.bgz" >chr 7 sites VCF (5.0 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr7.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr7.coverage.txt.gz" >chr 7 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr8.vcf.bgz" >chr 8 sites VCF (4.7 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr8.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr8.coverage.txt.gz" >chr 8 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr9.vcf.bgz" >chr 9 sites VCF (3.8 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr9.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr9.coverage.txt.gz" >chr 9 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr10.vcf.bgz" >chr 10 sites VCF (4.0 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr10.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr10.coverage.txt.gz" >chr 10 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr11.vcf.bgz" >chr 11 sites VCF (4.3 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr11.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr11.coverage.txt.gz" >chr 11 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr12.vcf.bgz" >chr 12 sites VCF (4.2 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr12.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr12.coverage.txt.gz" >chr 12 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr13.vcf.bgz" >chr 13 sites VCF (2.7 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr13.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr13.coverage.txt.gz" >chr 13 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr14.vcf.bgz" >chr 14 sites VCF (2.8 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr14.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr14.coverage.txt.gz" >chr 14 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr15.vcf.bgz" >chr 15 sites VCF (2.7 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr15.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr15.coverage.txt.gz" >chr 15 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr16.vcf.bgz" >chr 16 sites VCF (3.1 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr16.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr16.coverage.txt.gz" >chr 16 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr17.vcf.bgz" >chr 17 sites VCF (2.8 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr17.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr17.coverage.txt.gz" >chr 17 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr18.vcf.bgz" >chr 18 sites VCF (2.3 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr18.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr18.coverage.txt.gz" >chr 18 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr19.vcf.bgz" >chr 19 sites VCF (2.3 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr19.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr19.coverage.txt.gz" >chr 19 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr20.vcf.bgz" >chr 20 sites VCF (1.8 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr20.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr20.coverage.txt.gz" >chr 20 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr21.vcf.bgz" >chr 21 sites VCF (1.2 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr21.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr21.coverage.txt.gz" >chr 21 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr22.vcf.bgz" >chr 22 sites VCF (1.3 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chr22.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chr22.coverage.txt.gz" >chr 22 coverage</a>
                                    </p>
                                </td>
                            </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chrX.vcf.bgz" >chr X sites VCF (4.7 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.chrX.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/coverage/genomes/gnomad.genomes.r2.0.2.chrX.coverage.txt.gz" >chr X coverage</a>
                                    </p>
                                </td>
                            </tr>
                            <tr>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.coding_only.chr1-22.vcf.bgz" >Autosomes sites VCF (coding only, 2.4 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.coding_only.chr1-22.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                               <td width="428">
                                   <p>
                                       <a href="https://console.cloud.google.com/storage/browser/gnomad-public/release/2.0.2/coverage/genomes/" >Genome coverage</a>
                                   </p>
                               </td>
                           </tr>
                            <tr>
                                <td width="428">
                                    <p>
                                        <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.coding_only.chrX.vcf.bgz" >chr X sites VCF (coding only, 0.06 GB)</a> <a href="https://storage.googleapis.com/gnomad-public/release/2.0.2/vcf/genomes/gnomad.genomes.r2.0.2.sites.coding_only.chrX.vcf.bgz.tbi" >(.tbi)</a>
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        <a href="https://console.cloud.google.com/storage/browser/gnomad-public/release/2.0.2/coverage/genomes_exons_only/" >Genome coverage (exon calling regions only)</a>
                                    </p>
                                </td>
                           </tr>
                             <tr>
                                <td width="429">
                                    <p>
                                        &nbsp;
                                    </p>
                                </td>
                                <td width="428">
                                    <p>
                                        &nbsp;
                                    </p>
                                </td>
                            </tr>
                             <!-- <tr>
                                <td width="857">
                                    <p>
                                        <a href="https://data.broadinstitute.org/gnomAD/release/2.0.2/genomes/vcf/gnomad.genomes.r2.0.2.sites.1.vcf.bgz" >chr Y sites VCF</a>
                                    </p>
                                </td>
                            </tr> -->
                        </tbody>
                    </table>
                    </div>
                </div>
                <!-- <h2>Reference data</h2> -->
                <!-- <table class="downloads-table"border="1" cellspacing="0" cellpadding="0">
                    <tbody align="center">
                        <tr>
                            <td width="357">
                                <p>
                                    <strong>FTP Link</strong>
                                </p>
                            </td>
                            <td width="500">
                                <p align="center">
                                    <strong>Description</strong>
                                </p>
                            </td>
                        </tr>
                        <tr>
                            <td width="357">
                                <p>
                                    <a href="ftp://ftp.broadinstitute.org/pub/ExAC_release/release0.3.1/coverage" >Exome coverage</a>
                                </p>
                            </td>
                            <td width="500">
                                <p align="center">
                                    Exome Per Base Coverage
                                </p>
                            </td>
                        </tr>
                        <tr>
                            <td width="357">
                                <p>
                                    <a href="ftp://ftp.broadinstitute.org/pub/ExAC_release/release0.3.1/coverage" >Genome coverage</a>
                                </p>
                            </td>
                            <td width="500">
                                <p align="center">
                                    Genome Per Base Coverage
                                </p>
                            </td>
                        </tr>

                        <tr>
                            <td width="357">
                                <p>
                                    <a href="ftp://ftp.broadinstitute.org/pub/ExAC_release/release0.3.1/subsets" >Subsets</a>
                                </p>
                            </td>
                            <td width="500">
                                <p align="center">
                                    Non-TCGA VCF Subset
                                </p>
                            </td>
                        </tr>
                    </tbody>
                </table> -->
            </div>

    </div>
{% endblock %}

