{% extends "layout.html" %}
{% block loads %}
    <!-- jQuery and jQuery UI -->
    <link rel="stylesheet" type="text/css" href="//ajax.googleapis.com/ajax/libs/jqueryui/1.11.2/themes/smoothness/jquery-ui.css" />
    <script type="text/javascript" src="//ajax.googleapis.com/ajax/libs/jqueryui/1.11.2/jquery-ui.min.js"></script>

    <!-- Google Fonts and Font awesome -->
    <link rel="stylesheet" type="text/css" href='//fonts.googleapis.com/css?family=PT+Sans:400,700' />
    <link rel="stylesheet" type="text/css" href='//fonts.googleapis.com/css?family=Open+Sans' />
    <!-- ALREADY LOADED BY layout.html: link rel="stylesheet" type="text/css" href='//maxcdn.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.min.css' / -->
{% endblock %}
{% block body %}
    <!-- Render context vars in JS here -->
    <script type="text/javascript">
        window.exac = {{ exac|tojson|safe }};
        window.gnomAD = {{ gnomad|tojson|safe }};
        // console.log('window.exac.variant', window.exac.variant)
        // console.log('window.gnomAD.variant', window.gnomAD.variant)

        window.variant = gnomad.combineDataForVariantPage(
            window.exac.variant,
            window.gnomAD.variant
        )
        window.any_covered = {{ any_covered|tojson|safe }};
        window.base_coverage = {{ base_coverage|tojson|safe }};
        window.consequences = {{ consequences|tojson|safe }};
        window.metrics = {{ exac_metrics|tojson|safe }};
        function hasData(rawVariant) {
            if (rawVariant.variant_id) {
                return true
            }
            return false
        }

        $(document).ready(function() {
            $('[data-toggle="tooltip"]').tooltip();
            draw_region_coverage(window.base_coverage, 'mean', window.variant.ref);
            $('.coverage_metric_buttons').change(function () {
                var v = $(this).attr('id').replace('_covmet_button', '');
                $('.coverage_subcat_selectors').hide();
                if (v == 'covered') {
                    $('#over_x_select_container').show();
                    v = $('#over_x_select').val();
                } else {
                    $('#average_select_container').show();
                    v = $("#average_select").val();
                }
                draw_region_coverage(window.base_coverage, v, window.variant.ref);
            });
            $('#over_x_select').change(function () {
                draw_region_coverage(window.base_coverage, $(this).val(), window.variant.ref);
            });
            $('#average_select').change(function () {
                draw_region_coverage(window.base_coverage, $(this).val(), window.variant.ref);
            });

            function setInitialPopulationTablesState(exomeFilter, genomeFilter) {
                // console.log(exomeFilter, genomeFilter)
                console.log(window.variant)
                if (exomeFilter === 'PASS' && genomeFilter === 'PASS') {
                    return 'all'
                }
                if (exomeFilter === 'PASS') {
                    $('#genome_checkbox').attr('checked', false)
                    if (genomeFilter === 'No variant') {
                        $('#genome_checkbox_label').html('Genomes (no variant)')
                        $('#exome_checkbox').attr('disabled', true)
                        $('#genome_checkbox').attr('disabled', true)
                    } else {
                        $('#genome_checkbox_label').html('Genomes (filtered)')

                    }
                    return 'exac'
                }
                if (genomeFilter === 'PASS') {
                    $('#exome_checkbox').attr('checked', false)
                    if (exomeFilter === 'No variant') {
                        $('#exome_checkbox_label').html('Exomes (no variant)')
                        $('#exome_checkbox').attr('disabled', true)
                        $('#genome_checkbox').attr('disabled', true)
                    } else {
                        $('#exome_checkbox_label').html('Exomes (filtered)')
                    }
                    return 'gnomad'
                }
                if (exomeFilter !== 'PASS' && exomeFilter !== 'No variant') {
                    if (genomeFilter === 'No variant') {
                        $('#exome_checkbox_label').html('Exomes (filtered)')
                        $('#genome_checkbox_label').html('Genomes (no variant)')
                        $('#exome_checkbox').attr('disabled', true)
                        $('#genome_checkbox').attr('disabled', true)
                        return 'exac'
                    } else {
                        $('#exome_checkbox_label').html('Exomes (filtered)')
                        $('#genome_checkbox_label').html('Genomes (filtered)')
                        return 'all'
                    }
                }
                if (genomeFilter !== 'PASS' && genomeFilter !== 'No variant') {
                    if (exomeFilter === 'No variant') {
                        $('#genome_checkbox_label').html('Genomes (filtered)')
                        $('#exome_checkbox_label').html('Exomes (no variant)')
                        $('#exome_checkbox').attr('disabled', true)
                        $('#genome_checkbox').attr('disabled', true)
                        return 'gnomad'
                    } else {
                        $('#exome_checkbox_label').html('Exomes (filtered)')
                        $('#genome_checkbox_label').html('Genomes (filtered)')
                        return 'all'
                    }
                }
            }

            function getDataSelection(exomeState, genomeState) {
                if (exomeState && genomeState) {
                    $('#genome_checkbox').attr('disabled', false)
                    $('#exome_checkbox').attr('disabled', false)
                    return 'all'
                }
                if (exomeState && !genomeState) {
                    $('#exome_checkbox').attr('disabled', true)
                    return 'exac'
                }
                if (!exomeState && genomeState) {
                    $('#genome_checkbox').attr('disabled', true)
                    return 'gnomad'
                }
            }

            function updatePopulationTable(variant, selection) {
	        var ac_total = 0
	        var an_total = 0
	        var hom_total = 0
	        var hemi_total = 0
            console.log(selection)
                for (var pop in variant['all'].pop_acs) {
	            var has_data_for_pop = variant[selection].pop_acs && (pop in variant[selection].pop_acs)
                    ix = pop.replace(/[\s()]/g, "")

	            var pop_ac = has_data_for_pop ? variant[selection].pop_acs[pop] : 0
                    $('#table-count-'+ix).html(pop_ac)
	            ac_total += pop_ac

	            var pop_an = has_data_for_pop ? variant[selection].pop_ans[pop] : 0
                    $('#table-number-' + ix).html(pop_an)
	            an_total += pop_an

	            var has_data_for_pop_homs = variant[selection].pop_homs && (pop in variant[selection].pop_homs)
	            var pop_hom = has_data_for_pop_homs ? variant[selection].pop_homs[pop] : 0
                    $('#table-homozygotes-' + ix).html(pop_hom)
	            hom_total += pop_hom;

                    if (variant.chrom === 'X' || variant.chrom === 'Y') {
   	              var has_data_for_pop_hemis = variant[selection].pop_hemis && (pop in variant[selection].pop_hemis)
	             //  console.log(selection, has_data_for_pop_hemis, variant[selection]);
	              var pop_hemi = has_data_for_pop_hemis ? variant[selection].pop_hemis[pop] : 0
                      $('#table-hemizygotes-' + ix).html(pop_hemi)
	              hemi_total += pop_hemi;
                    }

                    var frequency = pop_an > 0 ? (pop_ac / pop_an).toPrecision(4) : 'NA'
                    $('#table-frequency-' + ix).html(frequency)

                }
                $('#table-count-sum').html(ac_total)
                $('#table-number-sum').html(an_total)
                $('#table-homozygotes-sum').html(hom_total)

                if (variant.chrom === 'X' || variant.chrom === 'Y') {
                  $('#table-hemizygotes-sum').html(hemi_total)
                }
                $('#table-total-freq').html( an_total > 0 ? (ac_total / an_total).toPrecision(4) : 'NA' )
                if (!variant.all.pop_acs['Ashkenazi Jewish']
                     || variant.all.pop_acs['Ashkenazi Jewish']  === 0) {
                    $('#ibd-footer').hide()
                    $('#asj').html('Ashkenazi Jewish')
                }
            }

            $('.population_selector_checkbox_container').change(function() {
                var exomeState = $('#exome_checkbox').is(":checked")
                var genomeState = $('#genome_checkbox').is(":checked")
                var dataSelection = getDataSelection(exomeState, genomeState)
                updatePopulationTable(window.variant, dataSelection)
            })

            var filter_definitions = {
                SEGDUP: 'Filtered variant: found in a segmental duplication region',
                LCR: 'Filtered variant: found in a low complexity region',
                InbreedingCoeff: 'Filtered variant: has an inbreeding coefficient < -0.3',
                RF: 'Filtered variant: failed random forests filters',
                AC0: 'Filtered variant: allele count is zero (i.e. no high-confidence genotype)'
            }

            function addFilterFlag(filter) {
                if (filter === 'No variant') {
                    return '<span class="label label-danger">No variant</span>'
                }

                if (filter !== 'PASS') {
                    var labels = filter.split('|').map(function(f) {
                        return '<span style="cursor: pointer;" class="label filter-label label-warning data-toggle="tooltip" delay="0" title="' + filter_definitions[f] + '">' + f + '</span> &#xa'
                    }).join(' ')
                    return '<div>' + labels + '</div>'
                }
                return '<span class="label label-success">Pass</span>'
            }
            $('#filter-status-exome').html(addFilterFlag(variant.exac.filter))
            $('#filter-status-genome').html(addFilterFlag(variant.gnomad.filter))

            function renderAlleleFrequency(allele_frequency) {
                var frequency = Number(allele_frequency)
                if (frequency === 0) {
                    return 0
                }
                else if (frequency >= 0.0001) {
                    return frequency.toPrecision(4)
                } else {
                    return Number(frequency.toPrecision(4)).toExponential()
                }
            }

            if (variant.exac.allele_num) {
                $('#ac-exome').html(variant.exac.allele_count)
                $('#an-exome').html(variant.exac.allele_num)
                $('#af-exome').html(renderAlleleFrequency(variant.exac.allele_freq))
            }
            if (variant.gnomad.allele_num) {
                $('#ac-genome').html(variant.gnomad.allele_count)
                $('#an-genome').html(variant.gnomad.allele_num)
                $('#af-genome').html(renderAlleleFrequency(variant.gnomad.allele_freq))
            }
            $('#ac-total').html(variant.all.allele_count)
            $('#an-total').html(variant.all.allele_num)
            $('#af-total').html(renderAlleleFrequency(variant.all.allele_freq))

            var initialSelection = setInitialPopulationTablesState(variant.exac.filter, variant.gnomad.filter)
            updatePopulationTable(window.variant, initialSelection)
        });
    </script>
    <style>
    .d3_graph {
        font: 10px sans-serif;
    }

    .bar rect {
        fill: steelblue;
        shape-rendering: crispEdges;
    }

    .bar text {
        fill: #fff;
    }

    .axis path, .axis line {
        fill: none;
        stroke: #000;
        shape-rendering: crispEdges;
    }
    </style>
    <script>
        var af_buckets = [0.00005, 0.0001, 0.0002, 0.0005, 0.001, 0.002, 0.005, 0.01, 0.02, 0.05, 0.1, 0.2, 0.5, 1];
        function get_af_bucket_text(bin) {
            if (bin == 'singleton' || bin == 'doubleton') {
                return 'This is the site quality distribution for all ' + bin + 's in ExAC.';
            } else if (bin == '0.00005') {
                return 'This is the site quality distribution for all variants with AF < ' + bin + ' in ExAC.';
            } else {
                return 'This is the site quality distribution for all variants with ' + af_buckets[af_buckets.indexOf(parseFloat(bin)) - 1] + ' < AF < ' + bin + ' in ExAC.';
            }
        }
        $(document).ready(function() {
            $('.frequency_display_buttons').change(function() {
                $('.frequency_displays').hide();
                var v = $(this).attr('id').replace('_button', '');
                $('#' + v + '_container').show();
            });

            $('#frequency_table').tablesorter({
                stringTo: 'bottom',
                sortList: [[4,1], [0,0]],
                headers: {
                    4: {
                        sorter: 'digit'
                    }
                }
            });

            function renderQualityHistogram() {
                if (window.exac.metrics && window.gnomAD.metrics) {
                    var genotypeDataSelection = 'exac'
                    var siteDataSelection = 'exac'
                } else if (window.exac.metrics && !window.gnomAD.metrics) {
                    var genotypeDataSelection = 'exac'
                    var siteDataSelection = 'exac'
                    $('input[name=genotype-source][value=gnomad]').attr('disabled', true)
                    $('input[name=site-source][value=gnomAD]').attr('disabled', true)
                } else if (!window.exac.metrics && window.gnomAD.metrics) {
                    var genotypeDataSelection = 'gnomad'
                    var siteDataSelection = 'gnomAD'
                    $('input[name=genotype-source][value=exac]').attr('disabled', true)
                    $('input[name=site-source][value=exac]').attr('disabled', true)
                    $('input[name=genotype-source][value=gnomad]').attr('checked', true)
                    $('input[name=site-source][value=gnomAD]').attr('checked', true)
                }
                if (window.variant != null && 'genotype_depths' in window.variant[genotypeDataSelection]) {
                    draw_quality_histogram(window.variant[genotypeDataSelection].genotype_depths[1], '#quality_display_container', false, 'Depth', 'Variant Carriers');
                    $('#quality_metrics_container').change(function() {
                        var genotypeDataSelection = $('input[name=genotype-source]:checked').val()
                        var depthOrQuality = $('input[name=depth-or-quality]:checked').val()
                        var carriersOrAll = Number($('input[name=carriers-or-all]:checked').val())
                        var ylab = carriersOrAll ? 'Variant Carriers': 'All Individuals'
                        var xlab = depthOrQuality === 'genotype_depths' ? 'Depth' : 'Genotype Quality'
                        draw_quality_histogram(window.variant[genotypeDataSelection][depthOrQuality][carriersOrAll], '#quality_display_container', false, xlab, ylab);
                    });

                    // Quality metric histograms
                    var data = _.zip(_.map(window[siteDataSelection].metrics['Site Quality']['mids'], Math.exp), window[siteDataSelection].metrics['Site Quality']['hist']);
                    draw_quality_histogram(data, '#quality_metric_container', true, 'Site Quality', 'Variants');
                    var bin = window[siteDataSelection].metrics['Site Quality']['metric'].split('_')[1];
                    $('#site_quality_note').html(get_af_bucket_text(bin));
                    var position = window[siteDataSelection].variant.site_quality
                    $('#site-quality-score').html('Site Quality: ' + position)
                    for (metricType in window[siteDataSelection].variant.quality_metrics) {
                        var metricValue = window[siteDataSelection].variant.quality_metrics[metricType]
                        $('#' + metricType).html(metricType + ': ' + metricValue)
                    }
                    add_line_to_quality_histogram(data, position, '#quality_metric_container', true);
                    var log_these = ['Site Quality', 'DP'];
                    $('#quality_collapse').change(function() {
                        var siteDataSelection = $('input[name=site-source]:checked').val()
                        var position = window[siteDataSelection].variant.site_quality
                        $('#site-quality-score').html('Site Quality: ' + position)
                        for (metricType in window[siteDataSelection].variant.quality_metrics) {
                            var metricValue = window[siteDataSelection].variant.quality_metrics[metricType]
                            $('#' + metricType).html(metricType + ': ' + metricValue)
                        }
                        var selectedMetric = $('#quality_metric_select').val()
                        var metricValue = window[siteDataSelection].variant.quality_metrics[selectedMetric]
                        if (selectedMetric.split(':')[1]) {
                            selectedMetric = 'Site Quality'
                            metricValue = window[siteDataSelection].variant.site_quality
                        }
                        var v = $('#quality_metric_select').val().split(': ');
                        var log = false;
                        // console.log('site quality data changed', siteDataSelection)
                        // console.log('selectedMetric:', selectedMetric, 'metricValue:', metricValue)
                        $('#site_quality_note').html('');
                        var data;
                        if (log_these.indexOf(v[0]) > -1) {
                            data = _.zip(_.map(window[siteDataSelection].metrics[v[0]]['mids'], Math.exp), window[siteDataSelection].metrics[v[0]]['hist']);
                            log = true;
                        } else {
                            data = _.zip(window[siteDataSelection].metrics[v[0]]['mids'], window[siteDataSelection].metrics[v[0]]['hist']);
                        }
                        if (v[0] == 'Site Quality') {
                            var bin = window[siteDataSelection].metrics['Site Quality']['metric'].split('_')[1];
                            $('#site_quality_note').html(get_af_bucket_text(bin));
                        }
                        // console.log(siteDataSelection, data)
                        draw_quality_histogram(data, '#quality_metric_container', log, v[0], 'Variants');
                        add_line_to_quality_histogram(data, metricValue, '#quality_metric_container', log);
                    });
                } else {
                    $('#quality_metrics_container').hide();
    {#                $('#frequency_info_container').hide();#}
                }
            }
            // var exomeState = $('#exome_checkbox').is(":checked")
            // var genomeState = $('#genome_checkbox').is(":checked")
            // var dataSelection = getDataSelection(exomeState, genomeState)

            renderQualityHistogram()
        });
    </script>
    <div class="container">
        <div class="row">
            <div class="col-md-6">
                <h1><span class="hidden-xs">Variant: </span>{{ variant.chrom }}:{{ variant.pos }} {{ variant.ref }} / {{ variant.alt }}</h1>
                {% if variant.mnps %}
                    {% for mnp in variant.mnps %}
                        {% if mnp.category == 'UNCHANGED' %}
                            <span class="label label-info">Note:</span>
                        {% else %}
                            <h5><span class="label label-warning">Warning!</span>
                        {% endif %}
                        This variant is found in phase with <a href="/variant/{{ mnp.site2 }}">{{ mnp.site2 }}</a>
                        {% if mnp.site3 %}
                            and <a href="/variant/{{ mnp.site3 }}">{{ mnp.site3 }}</a>
                        {% endif %}
                        in {{ mnp.number_samples }} individual{% if mnp.number_samples > 1 %}s{% endif %}{% if mnp.category != 'UNCHANGED' %}, altering its functional interpretation</h5>{% endif %}.
                    {% endfor %}
                {% endif %}
            </div>
            <div class="col-md-6">
            {% if variant.orig_alt_alleles|length > 1 %}
                <h5><span class="label label-info">Note:</span> This variant is multiallelic! The other alt alleles are:</h5>
                <ul>
                    {% for allele in variant.orig_alt_alleles %}
                        {% if allele != variant.variant_id %}
                            <li>
                                <a href="/variant/{{ allele }}">{{ allele }}</a>
                            </li>
                        {% endif %}
                    {% endfor %}
                </ul>
            {% endif %}
            </div>
        </div>
        <hr/>
{#      Upper display #}
        {% if variant.variant_id %}
            {% if variant.allele_num < 97129.6 %}
                <p><span class="label label-warning">Warning!</span> This variant is only covered in {{ (variant.allele_num/2)|int }} individuals (adjusted allele number = {{ variant.allele_num }}).<br/>
                This means that the site is covered in fewer than 80% of the individuals in ExAC, which may indicate a low-quality site.</p>
                <hr/>
            {% endif %}
            <div class="row">
                <div class="col-md-6">
                    <dl class="dl-horizontal" style="margin-bottom: 0px;">
                        <dt  class="upper-display-dt"></dt>
                        <dd class="upper-display-row">
                            <div class="upper-display-cell bold">Exomes</div>
                            <div class="upper-display-cell bold">Genomes</div>
                            <div class="upper-display-cell bold">Total</div>
                        </dd>
                        <dt class="upper-display-dt">Filter</dt>
                        <dd class="upper-display-row">
                            <div class="upper-display-cell" id="filter-status-exome"></div>
                            <div class="upper-display-cell" id="filter-status-genome"></div>
                        </dd>
                        <dt class="upper-display-dt">Allele Count</dt>
                        <dd class="upper-display-row">
                            <div class="upper-display-cell" id="ac-exome"></div>
                            <div class="upper-display-cell" id="ac-genome"></div>
                            <div class="upper-display-cell" id="ac-total"></div>
                        </dd>
                        <dt class="upper-display-dt">Allele Number</dt>
                        <dd class="upper-display-row">
                            <div class="upper-display-cell" id="an-exome"></div>
                            <div class="upper-display-cell" id="an-genome"></div>
                            <div class="upper-display-cell" id="an-total"></div>
                        </dd>
                        <dt class="upper-display-dt">Allele Frequency</dt>
                        <dd class="upper-display-row">
                            <div class="upper-display-cell" id="af-exome"></div>
                            <div class="upper-display-cell" id="af-genome"></div>
                            <div class="upper-display-cell" id="af-total"></div>
                        </dd>
                        <dt>dbSNP</dt>
                        {% if variant.rsid and variant.rsid != "." %}
                            <dd><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs={{ variant.rsid }}" target="_blank">{{ variant.rsid }}</a></dd>
                        {% else %}
                            <dd>Not found in dbSNP</dd>
                        {% endif %}
                        <dt>UCSC</dt>
                        <dd>
                            <a href="http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg19&highlight=hg19.chr{{ variant.chrom }}%3A{{ variant.pos }}-{{ variant.pos + variant.ref|length - 1 }}&position=chr{{ variant.chrom }}%3A{{ variant.pos - 25 }}-{{ variant.pos + variant.ref|length - 1 + 25 }}" target="_blank">
                                {{ variant.variant_id }}
                                <i class="fa fa-external-link"></i>
                            </a>
                        </dd>
                        <dt>ClinVar</dt>
                        <dd>
                            {% if not variant.rsid or variant.rsid == "." %}
                                <a href="http://www.ncbi.nlm.nih.gov/clinvar?term=({{ variant.chrom }}%5BChromosome%5D)%20AND%20{{ variant.pos }}%5BBase%20Position%20for%20Assembly%20GRCh37%5D" target="_blank">
                            {% else %}
                                <a href="http://www.ncbi.nlm.nih.gov/clinvar?term={{ variant.rsid }}%5BVariant%20ID%5D" target="_blank">
                            {% endif %}
                                Click to search for variant in Clinvar
                                <i class="fa fa-external-link"></i>
                            </a>
                        </dd>
                    </dl>
                </div>
                <div class="col-md-6">
                    <div class="row">
                        <div class="col-md-8 col-md-offset-1">
                            <div class="panel-group" id="metrics_outer_accordion" role="tablist" aria-multiselectable="true">
                                <div class="panel panel-default">
                                    <div class="panel-heading" role="tab" id="gt_quality_collapse_heading"   style="height: auto">
                                        <div class="panel-title">
                                            <a data-toggle="collapse" data-parent="#accordion" href="#gt_quality_collapse" aria-expanded="false" aria-controls="gt_quality_collapse">
                                                <small>Genotype Quality Metrics</small>
                                            </a>
                                        </div>
                                    </div>
                                    <div id="gt_quality_collapse" class="panel-collapse collapse" role="tabpanel" aria-labelledby="gt_quality_collapse_heading">
                                        <div class="row">
                                            <div class="col-md-10 col-md-offset-1">
                                                <div id="quality_metrics_container">
                                                    <span class="btn-group" data-toggle="buttons" id="quality_full_site_button_group">
                                                        <button class="btn btn-primary btn-sm active quality_full_site_buttons" id="variant_carriers_button"
                                                                data-tooltip="Show metric for only individuals with this allele.">
                                                            <input type="radio" checked value="1" name="carriers-or-all"> Variant carriers
                                                        </button>
                                                        <button class="btn btn-primary btn-sm quality_full_site_buttons" id="variant_site_button"
                                                                data-tooltip="Show metric for all individuals (at this site, whether or not they have this allele).">
                                                            <input type="radio" checked value="0" name="carriers-or-all"> All individuals
                                                        </button>
                                                    </span>
                                                    <div id="quality_display_container" class="d3_graph"></div>
                                                    <small><span class="label label-info">Note:</span> Plot may include low-quality genotypes that were excluded from allele counts in the table above</small>
                                                    <div class="genotype-quality-bottom-group genotype-quality">
                                                        <div class="btn-group quality_display_button_group" data-toggle="buttons">
                                                            <button class="btn btn-primary btn-sm active quality_display_buttons" id="genotype_depths_button"
                                                                    data-tooltip="Per sample depth. Capped at 100X">
                                                                <input type="radio" checked value="genotype_depths" name="depth-or-quality"> Depth</button>
                                                            <button class="btn btn-primary btn-sm quality_display_buttons" id="genotype_qualities_button"
                                                            value="genotype_qualities"
                                                            name="depth-or-quality"
                                                                    data-tooltip="Per sample genotype qualit.">
                                                                <input type="radio" value="genotype_qualities" name="depth-or-quality"> Genotype Quality
                                                            </button>
                                                        </div>
                                                        <form class="genotype-quality-data-selector-form" name="dataSelection">
                                                          <div class="dataRadio"><input checked type="radio" name="genotype-source" value="exac">Exomes</div>
                                                          <div class="dataRadio"><input type="radio" name="genotype-source" value="gnomad">Genomes</div>
                                                        </form>
                                                    </div>
                                                </div>
                                                <br/>
                                            </div>
                                        </div>
                                    </div>
                                </div>
                                <div class="panel panel-default">
                                    <div class="panel-heading" role="tab" id="quality_collapse_heading"   style="height: auto">
                                        <div class="panel-title">
                                            <a data-toggle="collapse" data-parent="#accordion" href="#quality_collapse" aria-expanded="false" aria-controls="quality_collapse">
                                                <small>Site Quality Metrics</small>
                                            </a>
                                        </div>
                                    </div>
                                    <div id="quality_collapse" class="panel-collapse collapse" role="tabpanel" aria-labelledby="quality_collapse_heading">
                                        <div class="row">
                                            <div class="col-md-10 col-md-offset-1 site-quality">
                                                <div id="quality_metric_container"></div>
                                                <form class="site-quality-data-selector-form quality-selector" name="dataSelection">
                                                  <div class="dataRadio"><input checked type="radio" name="site-source" value="exac">Exomes</div>
                                                  <div class="dataRadio"><input type="radio" name="site-source" value="gnomAD">Genomes</div>
                                                </form>
                                                <small><div id="site_quality_note"></div></small>
                                                <small><span class="label label-info">Note:</span> These are site-level quality metrics: they may be unpredictable for multi-allelic sites.</small>
                                                <select id="quality_metric_select" class="form-control">
                                                    <option id="site-quality-score" name="site-quality-choice"><small>Site Quality: {{ variant.site_quality }}</small></option>
                                                    {% for metric in variant.quality_metrics %}
                                                        <option name="site-quality-choice" id="{{ metric }}" value="{{ metric }}"><small>{{ metric }}: {{ variant.quality_metrics[metric] }}</small></option>
                                                    {% endfor %}
                                                </select>
                                                <br/>
                                            </div>
                                        </div>
                                    </div>
                                </div>
                                <a class="btn btn-info report" href="/report/exac/{{ variant.variant_id }}">Report this variant</a>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
            <hr/>
        {% endif %}
        <div class="row">
            <div class="col-md-6">
                <div id="annotation_container">
                {% if variant.variant_id %}
                    <div class="section_header">Annotations</div>
                    {% if variant.vep_annotations %}
                        <p>This variant falls on {{ variant.transcripts|length }} transcripts in {{ variant.genes|length }} genes:</p>
                        <div class="panel-group" id="annotation_accordion" style="margin-bottom: 0px;">
                            <div class="row">
                                {% if consequences|length > 1 %}
                                    <div class="col-md-6" style="border-right: 1px dashed #AAA;">
                                {% else %}
                                    <div class="col-md-6">
                                {% endif %}
                                {% for consequence in consequences.keys()[:(((consequences|length + 1) / 2)|int)] %}
                                    {% include 'variant_consequences.html' %}
                                {% endfor %}
                                </div>
                                <div class="col-md-6">
                                {% for consequence in consequences.keys()[(((consequences|length + 1) / 2)|int):(consequences|length)] %}
                                    {% include 'variant_consequences.html' %}
                                {% endfor %}
                                </div>
                            </div>
                        </div>
                        <small><span class="label label-info">Note:</span> This list may not include additional transcripts in the same gene that the variant does not overlap.</small>
                    {% else %}
                        No annotations were found for this variant.
                    {% endif %}
                {% else %}
                    <h3>Variant not found.</h3>
                {% endif %}
                </div>
                <div id="browser_container">
                </div>
            </div>

            <div class="col-md-6">
                {% if variant.pop_acs %}
                    <div id="frequency_info_container">
                        <div class="section_header">Population Frequencies</div>
                        <div id="frequency_table_container" class="frequency_displays">
                            {% with chrom = variant.chrom %}
                            <table id="frequency_table">
                                <thead>
                                    <tr>
                                        <th>Population</th>
                                        <th>Allele Count</th>
                                        <th>Allele Number</th>
                                        {% if chrom != 'Y' %}
                                            <th>Number of Homozygotes</th>
                                        {% endif %}
                                        {% if chrom == 'X' or chrom == 'Y' %}
                                            <th>Number of Hemizygotes</th>
                                        {% endif %}
                                        <th>Allele Frequency</th>
                                    </tr>
                                </thead>
                                <tbody>
                                    {% for pop in variant.pop_acs %}
                                        <tr>
                                            {% if pop == 'Ashkenazi Jewish' %}
                                                <td id="asj">{{ pop }}*</td>
                                            {% else %}
                                                <td>{{ pop }}</td>
                                            {% endif %}
                                            <td id="table-count-{{ pop.replace(' ', '').replace('(', '').replace(')', '') }}"></td>
                                            <td id="table-number-{{ pop.replace(' ', '').replace('(', '').replace(')', '') }}"></td>
                                            <td id="table-homozygotes-{{ pop.replace(' ', '').replace('(', '').replace(')', '') }}"></td>
                                            {% if chrom == 'X' or chrom == 'Y' %}
                                                <td id="table-hemizygotes-{{ pop.replace(' ', '').replace('(', '').replace(')', '') }}"></td>
                                            {% endif %}
                                            <td id="table-frequency-{{ pop.replace(' ', '').replace('(', '').replace(')', '') }}"></td>
                                        </tr>
                                    {% endfor %}
                                </tbody>
                                <tfoot>
                                    <tr>
                                        <td><b>Total</b></td>
                                        <td><b id="table-count-sum"></b></td>
                                        <td><b id="table-number-sum"></b></td>
                                        {% if chrom != 'Y' %}
                                            <td><b id="table-homozygotes-sum"></b></td>
                                        {% endif %}
                                        {% if chrom == 'X' or chrom == 'Y' %}
                                            <td><b id="table-hemizygotes-sum"></b></td>
                                        {% endif %}
                                        <td><b id="table-total-freq"></b></td>
                                    </tr>
                                </tfoot>
                            </table>
                            {% endwith %}
                        </div>
                        <div class="population_selector_checkbox_container">
                            <div class="population_selector_checkbox_title">
                                Include:
                            </div>
                            <div class="population_selector_checkboxes">
                                <label>
                                    <input checked type="checkbox" id="exome_checkbox" value="">
                                    <div id="exome_checkbox_label" class="checkbox-label">Exomes</div>
                                </label>

                                <label class='hidden-xs'>
                                    <input checked type="checkbox" id="genome_checkbox" value="">
                                    <div id="genome_checkbox_label" class="checkbox-label">Genomes</div>
                                </label>
                            </div>
                        </div>
                        <div id="ibd-footer">* For detailed analysis of Ashkenazi Jewish frequency see the <a href="https://ibd.broadinstitute.org/variant/{{ variant.variant_id }}">IBD Exomes Browser</a>.</div>
                    </div>
                {% else %}
                    {% if any_covered %}
                        <div class="row">
                            <span class="section_header">Coverage</span>
                            {% if base_coverage|length > 1 %}
                                {% include 'coverage_selectors.html' %}
                            {% endif %}
                        </div>
                        <div id="region_coverage"></div>
                    {% else %}
                        <h3>This region is not covered.</h3>
                    {% endif %}
                {% endif %}
            </div>
        </div>

        {% if read_viz['total_expected'] > 0 %}

            <hr/>

            <div class="row">
                <div class="col-md-12">
                    <div class="section_header">Read Data</div>
                    <p/>
                    {% if read_viz['total_available'] == 0 %}
                    <h5 id='missing-data-note' style='margin-right:60%'>
                      <span class="label label-danger">Error:</span> &nbsp; Read data not available for this variant.
                    </h5>
                    <br><br>
                    {% else %}
                    This interactive <a href="https://github.com/igvteam/igv.js">IGV.js</a> visualization shows reads
                    that went into calling this variant.<br>
                    <small><span class="label label-info">Note:</span> These are reassembled reads produced by
                        <a href="https://www.broadinstitute.org/gatk/gatkdocs/org_broadinstitute_gatk_tools_walkers_haplotypecaller_HaplotypeCaller.php#--bamOutput">
                            GATK HaplotypeCaller --bamOutput
                        </a> so they accurately represent what HaplotypeCaller was seeing when it called this variant.
                    </small>
                    <br><br>

                    <div id="igv-container"></div>
                    <br><br>
                    <center>
 		    {% for exomes_or_genomes in ('exomes', 'genomes') %}
                      {% if read_viz[exomes_or_genomes]['het']['n_available'] > 0 or read_viz[exomes_or_genomes]['hom']['n_available'] > 0 or (variant.chrom in ('X', 'Y') and read_viz[exomes_or_genomes]['hemi'] > 0) %}
		        <div style="display:inline-block; min-width: 130px; vertical-align:bottom; font-weight:600; font-size:16px">{{ exomes_or_genomes.capitalize() }}:</div>
   		        <div style="display:inline-block;">

                        <button class="btn {% if exomes_or_genomes == 'exomes' %} btn-primary {% else %} btn-success {% endif %} quality_full_site_buttons"
                                style="disabled:true; width:130px; height: 32px;"
				id="load-more-{{ exomes_or_genomes }}-het-button">Load +1 Het
                        </button>
                        &nbsp; &nbsp; &nbsp;
                        <button class="btn {% if exomes_or_genomes == 'exomes' %} btn-primary {% else %} btn-success {% endif %} quality_full_site_buttons"
                                style="disabled:true; width:130px; height: 32px;"
				id="load-more-{{ exomes_or_genomes }}-hom-button">Load +1 Hom
                        </button>
			{% if variant.chrom in ('X', 'Y') %}
                        &nbsp; &nbsp; &nbsp;
                        <button class="btn {% if exomes_or_genomes == 'exomes' %} btn-primary {% else %} btn-success {% endif %} quality_full_site_buttons"
                                style="disabled:true; width:130px; height: 32px;"
				id="load-more-{{ exomes_or_genomes }}-hemi-button">Load +1 Hemi
                        </button>
			{% endif %}
			<br />
                      {% endif %}
		      </div>
		      <div style="display:inline-block; min-width: 180px;"> </div>
		      <br /><br />
		    {% endfor %}
                    </center>
                    <br>

                    <!-- IGV dependencies -->

                    <!-- igv.js -->
                    <!-- link rel="stylesheet" type="text/css" href="/igv-css/igv.css" -->
                    <!-- script type="text/javascript" src="/igv.min.js"></script -->

                    <link rel=stylesheet type=text/css href="{{ url_for('static', filename='css/igv.css') }}">
                    <script type="text/javascript" src="{{ url_for('static', filename='igv.min.js') }}"></script>

                    <!-- link rel="stylesheet" type="text/css" href="http://igv.org/web/beta/igv-beta.css" -->
                    <!-- script type="text/javascript" src="http://igv.org/web/beta/igv-beta.min.js"></script -->

                    <script type="text/javascript">

                        var get_bam_track_config = function (read_viz_data, exomes_or_genomes, het_or_hom_or_hemi, i) {
                            return {
                                type: 'bam',
                                indexed: true,
                                alignmentShading: 'strand',
                                url: '/read_viz/' + read_viz_data[exomes_or_genomes][het_or_hom_or_hemi]['urls'][i],
                                name: het_or_hom_or_hemi + ' [' + exomes_or_genomes.slice(0,-1) + '] #' + (i + 1),
                                height: 300,
                                minHeight: 300,
                                autoHeight: false,
                                readgroup: read_viz_data[exomes_or_genomes][het_or_hom_or_hemi]['readgroups'][i],  //readgroup: '1-157768000-G-C_hom10',
                            };
                        };

                        // global igv settings
                        igv.CoverageMap.threshold = 0.1;

                        // temporary implementation of filtering IGV BamTracks by a user-specified ReadGroup
                        igv.BamReader.prototype.readFeatures_original = igv.BamReader.prototype.readFeatures;
                        igv.BamReader.prototype.readFeatures = function (chr, min, max, continuation, task) {

                            //if a readgroup was passed to the BamTrack constructor, filter reads without this read group.
                            if (this.config.readgroup) {
                                var readgroup = this.config.readgroup;
                                var continuation_original = continuation;
                                var continuationNew = function (alignments) {
                                    //filter the alignments by read group
                                    //console.log("Filtering by RG == " + readgroup );
                                    alignments = alignments.filter(function (alignment) {
                                        alignment.tags(); //parses tags from their binary representation and populates the alignment.tagDict
                                        return alignment.tagDict['RG'] == readgroup;
                                    });

                                    //forward the filtered aligments to the original continuation
                                    return continuation_original(alignments);
                                };
                            }

                            return this.readFeatures_original(chr, min, max, continuationNew, task);
                        };

                        //a dict of counts: n_expected_het, n_expected_hom, n_available_het, n_available_hom
                        var read_viz_data = {{ read_viz | tojson | safe }};
			console.log(read_viz_data);

                        //counts number of het/hom bam tracks displayed so far
                        var tracks_counter = { 'exomes': {"het": 0, "hom": 0, "hemi": 0}, 'genomes': {"het": 0, "hom": 0, "hemi": 0}};

                        //initialize IGV tracks
                        var tracks = [];
                        tracks.push({
                            url: '/read_viz/gencode.v19.sorted.bed',
                            name: "gencode v19",
                            displayMode: "SQUISHED"
                        });
                        //tracks.push({ url: '/read_viz/exome_calling_regions.v1.bed', name: "ExAC calling regions" });
                        //tracks.push({ url: '/read_viz/self_chain.sorted.bed',        name: "UCSC self chain" });

                        if (location.hash == '#all') {
                            //add all available tracks for het and hom and hide the 'load +1' buttons
                            // this mode was requested by Konrad and Eric
                            ["exomes", "genomes"].forEach(function (genomes_or_exomes) {
                              ["het", "hom", "hemi"].forEach(function (het_or_hom_or_hemi) {
                                $('#load-more-'+genomes_or_exomes+'-'+het_or_hom_or_hemi+'-button').hide();
                                var n = read_viz_data[genomes_or_exomes][het_or_hom_or_hemi]['n_available'];
                                for (var i = 0; i < n; i += 1) {
                                    tracks.push(get_bam_track_config(read_viz_data, genomes_or_exomes, het_or_hom_or_hemi, i));
                                    tracks_counter[genomes_or_exomes][het_or_hom_or_hemi] += 1;
                                }
                              })
                            });

                        } else {
                            //add track #1 for het and hom (assuming they're available)
                            ["genomes", "exomes"].forEach(function (genomes_or_exomes) {
                              ["het", "hom", "hemi"].forEach(function (het_or_hom_or_hemi) {
                                if (read_viz_data[genomes_or_exomes][het_or_hom_or_hemi]['n_available'] > 0) {
                                    tracks.push(get_bam_track_config(read_viz_data, genomes_or_exomes, het_or_hom_or_hemi, 0));
                                    tracks_counter[genomes_or_exomes][het_or_hom_or_hemi] += 1;
                                }

                                disable_load_one_more = tracks_counter[genomes_or_exomes][het_or_hom_or_hemi] >= read_viz_data[genomes_or_exomes][het_or_hom_or_hemi]['n_available'];
                                $('#load-more-'+genomes_or_exomes+'-'+het_or_hom_or_hemi+'-button').prop('disabled', disable_load_one_more);
                                if(disable_load_one_more) {
                                    $('#load-more-'+genomes_or_exomes+'-'+het_or_hom_or_hemi+'-button').html('No more ' + het_or_hom_or_hemi);
                                }
                              });
                            });

                            var load_one_more = function (genomes_or_exomes, het_or_hom_or_hemi) {
                                if (tracks_counter[genomes_or_exomes][het_or_hom_or_hemi] < read_viz_data[genomes_or_exomes][het_or_hom_or_hemi]['n_available']) {
                                    var i = tracks_counter[genomes_or_exomes][het_or_hom_or_hemi];
                                    igv.browser.loadTrack(get_bam_track_config(read_viz_data, genomes_or_exomes, het_or_hom_or_hemi, i));
                                    tracks_counter[genomes_or_exomes][het_or_hom_or_hemi] += 1;
                                }

                                if (tracks_counter[genomes_or_exomes][het_or_hom_or_hemi] >= read_viz_data[genomes_or_exomes][het_or_hom_or_hemi]['n_available']) {
                                    $('#load-more-'+genomes_or_exomes+'-'+het_or_hom_or_hemi+'-button').prop('disabled', true);
                                    $('#load-more-'+genomes_or_exomes+'-'+het_or_hom_or_hemi+'-button').html('No more ' + het_or_hom_or_hemi);
                                }
                            };

                            ["genomes", "exomes"].forEach(function (genomes_or_exomes) {
                              ["het", "hom", "hemi"].forEach(function (het_or_hom_or_hemi) {
                                  $('#load-more-'+genomes_or_exomes+'-'+het_or_hom_or_hemi+'-button').click(function () {
                                     load_one_more(genomes_or_exomes, het_or_hom_or_hemi);
                                  });
                               })
                            })
                        }



                        //initialize IGV.js browser
                        var locus = '{{ variant.chrom }}:{{ variant.pos-40 }}-{{ variant.pos+40 }}';
                        var options = {
                            showCommandBar: true,
                            genome: 'hg19',
                            locus: locus,
                            showKaryo: false,
                            tracks: tracks,
                            id:'hg19',
                            fastaURL: '/read_viz/hg19.fa',
			    indexURL: '/read_viz/hg19.fa.fai'
                        };

                        igv.createBrowser($("#igv-container")[0], options);


                        //IGV browser customizations
                        $(".igv-ideogram-content-div").hide();  //hide the IGV ideogram

                        //$(".igv-logo").hide();

                        $(".igvNavigationSearch").append(
                                "<a id='reset-locus' title='Reset to original locus (" + locus + ")'>" +
                                "<i class='igv-app-icon fa fa-mail-reply shim-left-6'></i>" +
                                "</a>");

                        // allow one more zoom-in level
                        igv.browser.pixelPerBasepairThreshold = function () {
                            return 28.0;  //default is currently 14.0
                        };

                        //click handlers
                        $('#reset-locus').click(function () {
                            igv.browser.search(locus);
                        });

                        igv.browser.trackViews.forEach(function (panel) {
                            if (panel.track.name) {
                                //add border between tracks
                                panel.viewportDiv.style.borderBottom = panel.viewportDiv.style.borderLeft = "1px solid #cccccc";
                            }
                        });

                    </script>
                    <style>
                        .igv-viewport-div {
                            border-left: 1px solid #cccccc;
                            border-bottom: 1px solid #cccccc;
                        }
                        $(document).ready(function(){
    $('[data-toggle="tooltip"]').tooltip();
});
                    </style>
                    {% endif %}
                </div>
            </div>
	    {% endif %}
        </div>

{#% endif %#}
{% endblock %}
